google-cloud-spanner/src/main/java/com/google/cloud/spanner/connection/ConnectionImpl.java

```diff
@@ -623,6 +633,55 @@ public void setStatementTag(String tag) {
     this.statementTag = tag;
   }
 
+  @Override
+  public byte[] getProtoDescriptors() {
+    ConnectionPreconditions.checkState(!isClosed(), CLOSED_ERROR_MSG);
+    if (this.protoDescriptorsFilePath != null) {
+      // Read from file if filepath is valid
+      try {
+        File protoDescriptorsFile = new File(this.protoDescriptorsFilePath);
+        String fileName = protoDescriptorsFile.getName();
+        int dotIndex = fileName.lastIndexOf('.');
+        // check if the file is a .pb extension file.
+        if (dotIndex < 0
+            || dotIndex > fileName.length() - 1
+            || !fileName.substring(dotIndex + 1).equals("pb")) {
+          throw SpannerExceptionFactory.newSpannerException(
+              ErrorCode.INVALID_ARGUMENT, "Proto descriptor file should be a .pb extension file");
```

Comment:
Thanks for bringing this up, Knut. I was under the assumption that proto descriptor file must be a `.pb` file. To clarify that hypothesis, I did a small experiment now and it turned out my assumption was wrong. It can be any file and it worked with `.txt` as well.
This validation is no longer valid and hence removing it.